app/Services/CommentService.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace App\Services;
+
+use App\Models\Comment;
+
+class CommentService
+{
+    public function create($request)
+    {
+        return Comment::create([
+            'user_id' => $request->user()->id,
+            'post_id' => $request->input('postId'),
+            'content' => $request->input('content'),
+        ]);
+    }
+
+    public function getAll($postId)
+    {
+        return Comment::with('user')->where([
+            ['post_id', '=', $postId],
+        ])->orderBy('created_at', 'desc')->paginate(8);
+    }
+
+    public function getCommentQuantity($request)
+    {
+        return Comment::with('user')->where([
+            ['post_id', '=', $request->input('postId')],
+        ])->orderBy('created_at', 'desc')->count();
```

Comment:
Cái này chắc ko cần load user ra e nhỉ
```suggestion
        return Comment::where([
            ['post_id', '=', $request->input('postId')],
        ])->orderBy('created_at', 'desc')->count();
```

Reply:
@tuanpht dạ có chứ. 
Comment hiển thị ra có `[avatar] @username [Nội dung comment]` nên em phải load user ra để trả về ạ?

Reply:
Trong `CommentResource`:
```js
public function toArray($request)
    {
        return [
            ...
            'author' => new UserResource($this->whenLoaded('user')),
            ...
        ];
}
```

Reply:
Anh thấy cái này chỉ return count thôi, không phải model

Reply:
dạ vâng, hàm `getCommentQuantity` return ra count chứ có return ra model đâu ạ
```js
public function getCommentQuantity($request)
    {
        return Comment::with('user')->...->count();
}
```

Reply:
dạ vâng, hàm getCommentQuantity return ra count chứ có return ra model đâu ạ @tuanpt-0634 
```js
public function getCommentQuantity($request)
    {
        return Comment::with('user')->...->count();
}
```